CONTRIBUTING.md

```diff
@@ -0,0 +1,84 @@
+# Contributing
+
+## Getting Started
+
+General information about setting up your Python environment, testing modules,
+Ansible coding styles, and more can be found in the [Ansible Community Guide](
+https://docs.ansible.com/ansible/latest/community/index.html).
+
+
+## ServiceNow ITSM Collections
+
+### servicenow.itsm
+
+The Ansible Collection for ServiceNow IT Service Management ([ITSM](https://www.servicenow.com/products/itsm.html)) includes a variety of Ansible content to help automate the management of ServiceNow IT Service Management.
+
+New modules and plugins developed by the community should be proposed to `servicenow.itsm`.
+
+## Submitting Issues
+All software has bugs, and the `servicenow.itsm` collection is no exception. When you find a bug,
+you can help tremendously by [telling us about it](https://github.com/ansible-collections/servicenow.itsm/issues).
+
+If you should discover that the bug you're trying to file already exists in an issue,
+you can help by verifying the behavior of the reported bug with a comment in that
+issue, or by reporting any additional information.
+
+## Writing New Code
+
+New modules should be submitted to the [servicenow.itsm](https://github.com/ansible-collections/servicenow.itsm) collection.
+
+### Static analysis
+For new features and bug fixes on existing modules, clone this repository and try to run unit tests and integration tests by following [these instructions](https://docs.ansible.com/ansible/latest/community/create_pr_quick_start.html). When you get to this part, we use a wide variety of linters and static code analysis tools to ensure our code base stays healthy. To run all checks, we can execute the following command:
+
+```
+(venv) $ make sanity
+```
+This command will take care of installing requirements and running them.
+
+### Unit tests
+We use unit tests to test the logic and implementation of our helper functions and various Ansible plugins that are part of the `servicenow.itsm` collection.
+```
+(venv) $ make units
+```
+
+### Integration tests
```

Comment:
I think there needs to be something about ServiceNow and setting up their test environment for this? Not sure that is the case and if so that this is the right place for it. Thinking as a developer that is a question I have after reading this -- what about my ServiceNow setup to test?

Also should this doc mention anything about the ServiceNow app or at least link to something regarding that?

Reply:
For me, integration tests against free developer servicenow instance were super slow, especially when it arrived to configuration modules. The connection got interrupted from time to time for various reasons including SNOW instance hibernation in the middle of the test. It would be nice to have some tips how to make it work as good as possible.

The link to SNOW developer page may be also fine but I think that more details are not necessary. There are guides on SNOW pages how to set up SNOW testing environment. If this somehow enters documentation here, the changes on the SNOW side could outdate information here. I think reference to [General Usage Patterns](https://github.com/ansible-collections/servicenow.itsm/blob/main/docs/general_usage_patterns.md) is quite enough.

Reply:
I was planning on writing a quick guide on how you can Set up your ServiceNow instance and OAuth credentials in the CONTRIBUTING doc, and then reference to [General Usage Patterns](https://github.com/ansible-collections/servicenow.itsm/blob/main/docs/general_usage_patterns.md) for more information.